besu/src/main/java/org/hyperledger/besu/RunnerBuilder.java

```diff
@@ -590,7 +597,7 @@ public Runner build() {
     final NatMethod detectedNatMethod =
         Optional.of(natMethod)
             .filter(not(isEqual(NatMethod.AUTO)))
-            .orElse(NatService.autoDetectNatMethod(new DockerDetector()));
+            .orElse(NatService.autoDetectNatMethod(new DockerDetector(), new KubernetesDetector()));
```

Comment:
What happens if it would return the `NatMethod` for both of these? Do we want `Docker` to be returned even if `Kubernetes` would be detected?

Reply:
The first method detected would be the one selected. In the order they are declared. However in that case Kubernetes would be detected. The detection method for docker and kubernetes are mutually exclusive.

Reply:
> The detection method for docker and kubernetes are mutually exclusive.
I understand that only one would end up being selected but is it possible a misconfigured docker container inside a kubernetes "pod" (or whatever the term is haha), could succeed on the docker detector and not go on to the kubernetes detector? I ask because my intuition tells me that we should attempt to detect if we're in kubernetes first since it's the case that is a less general case.

Reply:
> > The detection method for docker and kubernetes are mutually exclusive.
> > I understand that only one would end up being selected but is it possible a misconfigured docker container inside a kubernetes "pod" (or whatever the term is haha), could succeed on the docker detector and not go on to the kubernetes detector? I ask because my intuition tells me that we should attempt to detect if we're in kubernetes first since it's the case that is a less general case.

Ok I can create a new PR in order to change the order and put kubernetes detector first.